client/repo.c

```diff
@@ -467,6 +467,155 @@ TDNFEventRepoMDDownloadEnd(
     goto cleanup;
 }
 
+uint32_t
+TDNFStoreBaseURLFromMetalink(
+    PTDNF pTdnf,
+    const char *pszRepo,
+    const char *pszRepoMDURL
+    )
+{
+    uint32_t dwError = 0;
+    char *pszBaseUrlFile = NULL;
+    PTDNF_REPO_DATA_INTERNAL pRepos = NULL;
+
+    if (!pTdnf ||
+       !pTdnf->pConf ||
+       IsNullOrEmptyString(pszRepo) ||
+       IsNullOrEmptyString(pszRepoMDURL))
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    if (!pTdnf->pRepos)
+    {
+        dwError = ERROR_TDNF_NO_REPOS;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+    pRepos = pTdnf->pRepos;
```

Comment:
nit: move this assignment above the previous if and modify the check accordingly.